Document manifest size limit recommendation (#293)
opencontainers/distribution-spec: spec.md

An attempt to pull a nonexistent repository MUST return response code `404 Not Found`

> A registry SHOULD enforce some limit on the maximum manifest size that it can accept.

Reviewer: While this is recommended for public registries, I can see a case for private registries not wanting to implement this.

```suggestion
A registry MAY enforce limits on the maximum manifest size that it can accept.
```

A registry that enforces this limit SHOULD respond to a request to push a manifest over this limit with a response code `413 Payload Too Large`.

> Client and registry implementations SHOULD expect to be able to support manifest pushes of at least 4 megabytes.

Reviewer: I'd suggest splitting this into two lines, one for clients, and one for registries:

```suggestion
A registry that implementations this SHOULD support manifest pushes of at least 4 megabytes.
A client SHOULD avoiding pushing a manifest larger than 4 megabytes and SHOULD support pulling a manifest of at least 4 megabytes.
```

Reviewer 2: Maybe something like:

> Registries SHOULD NOT impose a limit less than 4 megabytes on the size of a manifest.
> Clients SHOULD support handling manifest sizes of at least 4 megabytes.
> Clients concerned with portability SHOULD NOT generate manifests larger than 4 megabytes.

FWIW, if quay has a 1MB limit, I would lean toward 1MB as the recommendation.

Reviewer 3: The wording from @jonjohnsonjr is more clear.
